File: trunk/quote/src/misc/UOMConverter.vb
Imports System.ComponentModel

Public Class UOMConverter : Inherits StringConverter

    Public Overloads Overrides Function _
        GetStandardValues(ByVal context As  _
        System.ComponentModel.ITypeDescriptorContext) _
        As TypeConverter.StandardValuesCollection

        Dim l As New List(Of String)

        Dim table As QuoteDataBase._UnitOfMeasureDataTable
        table = New QuoteDataBaseTableAdapters._UnitOfMeasureTableAdapter().GetData()
        For Each row As QuoteDataBase._UnitOfMeasureRow In table.Rows
            l.AddRange(New String() {row.Name })
        Next
        l.Sort()

        Return New StandardValuesCollection(l)
    End Function

    Public Overloads Overrides Function _
        GetStandardValuesExclusive(ByVal context _
        As System.ComponentModel.ITypeDescriptorContext) _
        As Boolean

        Return False
    End Function

    Public Overrides Function _
        GetStandardValuesSupported(ByVal context As ITypeDescriptorContext) As Boolean

        Return True
    End Function

End Class

File: quote/Projects/PluginModel/UOMConverter.cs
namespace Model
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Linq;

    public class UnitOfMeasureConverter : StringConverter
    {
        public override TypeConverter.StandardValuesCollection GetStandardValues(System.ComponentModel.ITypeDescriptorContext context)
        {
            List<string> l = new List<string>();

            DB.QuoteDataBase._UnitOfMeasureDataTable table = null;
            table = new DB.QuoteDataBaseTableAdapters._UnitOfMeasureTableAdapter().GetData();
            foreach (DB.QuoteDataBase._UnitOfMeasureRow row in table.Rows) 
            {
                l.AddRange(new string[] { row.Name });
            }
            
            l.Sort();

            return new StandardValuesCollection(l);
        }

        public override bool GetStandardValuesExclusive(System.ComponentModel.ITypeDescriptorContext context)
        {
            return false;
        }

        public override bool GetStandardValuesSupported(ITypeDescriptorContext context)
        {
            return true;
        }
    }
}
